test/service/FileTransferService.test.ts

```diff
@@ -0,0 +1,69 @@
+import { expect } from 'chai';
+import FormData from 'form-data';
+import nock = require('nock');
+
+import { FileTransferService } from 'app/service/FileTransferService'
+
+describe('FileTransferService', () => {
+
+    const KEY: string = 'mockKey';
+    const HOST: string = 'http://localhost';
+    const evidenceUploadService = new FileTransferService(HOST , KEY);
+
+    describe('upload file', () => {
+        const URI: string = '/dev/files';
+
+        it('should throw an error when evidence not defined', () => {
+
+            [undefined, null].forEach(async evidence => {
+                try {
+                    await evidenceUploadService.upload(evidence!, 'filename')
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('Evidence file is missing')
+                }
+            })
+        });
+
+        it('should throw an error when file name not defined', () => {
+
+            const evidence: Buffer = Buffer.from('This is a test');
+
+            [undefined, null].forEach(async filename => {
+                try {
+                    await evidenceUploadService.upload(evidence as Buffer, filename!)
+                } catch (err) {
+                    expect(err).to.be.instanceOf(Error)
+                        .and.to.haveOwnProperty('message').equal('File name is missing')
+                }
+            })
+        });
+
+
+        it('should return 201 status code when evidence uploaded', async () => {
+
+            const EVIDENCE_ID: string = 'Mock ID';
+            const evidence: Buffer = Buffer.from('This is a test');
+
+            const data = new FormData();
+            data.append('upload', evidence, {filename: 'test'});
+
+            nock(HOST)
+                .post(URI,
+                    new RegExp(`form-data; name="upload"[^]*${evidence}`,'m'),
```

Comment:
This is how API call looks like from the HTTP point of view:

```
00c0: Content-Type: multipart/form-data; boundary=--------------------
0100: ----f62dabfe821ce10a
0116:
== Info: Connection state changed (MAX_CONCURRENT_STREAMS == 128)!
=> Send data, 33605 bytes (0x8345)
0000: --------------------------f62dabfe821ce10a
002c: Content-Disposition: form-data; name="upload"; filename="light-9
006c: 4a07e06a1-v2 (1).woff2"
0085: Content-Type: application/octet-stream
```

Reply:
It might be clearer to do matching based on file name e.g.

```
new RegExp(`form-data; name="upload"; filename="..."`, 'm')
```

Reply:
But I do not know the filename of evidence